### ISSUE TITLE
Too many TCP Ports

### ISSUE THREAD

**Issue by CWolffIF** [eXpandFramework/eXpand#891]
Hi,

using an XAF Application long time it opens many TCP ports and do not close them.

I tried with a simple XAF project without eXpandFramwork. There it is OK. But if I add the expandFramework the error occurs and many connections are open.
I sent you the DEMO project.

![image](https://user-images.githubusercontent.com/43132306/139819223-af54722e-56dd-40fc-850a-fa95847c4473.png)

![image](https://user-images.githubusercontent.com/43132306/139819100-252ae4d0-3740-4a9d-aad5-003a9dcb55a1.png)

[DXApplication2.zip](https://github.com/eXpandFramework/eXpand/files/7459357/DXApplication2.zip)

**Comment by CWolffIF** [comment]
The problem occurs a you register
RequiredModuleTypes.Add(typeof(Xpand.ExpressApp.SystemModule.XpandSystemModule));
            RequiredModuleTypes.Add(typeof(Xpand.ExpressApp.Security.XpandSecurityModule));

**Comment by apobekiaris** [comment]
hmm i fail to repro it

```ps1
$id=(Get-Process -Name DXApplication2.Win).Id
netstat /a /n /o | find " $id"
```

or 

![image](https://user-images.githubusercontent.com/159464/139826883-f4535f7e-7b7c-41db-b1e3-569a7182ab81.png)

only the Reactive.Logger.Hub opens ports and that package is not a dependecy to anyone. If you have better instructions let me know

**Comment by CWolffIF** [comment]
Hi,
are you sure you using TCP/IP for the SQL Connection? (SQL Server and Application on the same machine often do not use TCP/IP and in your Screenshot you do not have a TCP Connection, maybe Named Pipes???)

In my sample

![image](https://user-images.githubusercontent.com/43132306/139828148-72d27943-5321-43c7-ab1d-1d249e3b5403.png)

**Comment by apobekiaris** [comment]
right applogies 1433 escaped me

**Comment by apobekiaris** [comment]
We will try to answer all questions that do not require research within 24hr.
To prioritize cases that require research we use the following labels **in order**. For all other issues the posting time is respected.
1. [❇️ Exclusive-Services](https://github.com/eXpandFramework/eXpand/labels/❇️%20Exclusive-Services)
 1. [❤ Bronze Sponsor](https://github.com/eXpandFramework/eXpand/labels/❤%20Bronze%20Sponsor)
 1. [❤ Sponsor](https://github.com/eXpandFramework/eXpand/labels/❤%20Sponsor)
 1. [❤ Backer](https://github.com/eXpandFramework/eXpand/labels/❤%20Backer)
 1. [Installation](https://github.com/eXpandFramework/eXpand/labels/Installation)
 1. [ShowStopper](https://github.com/eXpandFramework/eXpand/labels/ShowStopper)
 1. [Nuget](https://github.com/eXpandFramework/eXpand/labels/Nuget)
 1. [Contribution](https://github.com/eXpandFramework/eXpand/labels/Contribution)
 1. [BreakingChange](https://github.com/eXpandFramework/eXpand/labels/BreakingChange)
 1. [ReproSample](https://github.com/eXpandFramework/eXpand/labels/ReproSample)
 1. [Deployment](https://github.com/eXpandFramework/eXpand/labels/Deployment)
 1. [must-have](https://github.com/eXpandFramework/eXpand/labels/must-have)


**This case is prioritized.**

**Comment by apobekiaris** [comment]
found what to blame is the `SequenceGeneratorModule)` and to workaround u have to unregister this module

```cs
ReactiveModuleBase.Unload(typeof(SequenceGeneratorModule));
```

I will provide a solution to the problem 21.2.3 thnks for reporting it

**Comment by vimarx** [comment]
> found what to blame is the `SequenceGeneratorModule)` and to workaround u have to unregister this module
> 
> ```cs
> ReactiveModuleBase.Unload(typeof(SequenceGeneratorModule));
> ```
> 
> I will provide a solution to the problem 21.2.3 thnks for reporting it

Where exactly do I have to unload it?

**Comment by apobekiaris** [comment]
any place before application setup

**Comment by CWolffIF** [comment]
Oh, we need that module.

**Comment by apobekiaris** [comment]
I try my best this a major upgrade and endless test needs to pass before release. Cannot patch just this module unfortunately

**Comment by CWolffIF** [comment]
Ok, we wait and hope. Good luck.

**Comment by isatahiri** [comment]
Hi,

Is the SequenceGeneratorModule leaving the connections open ? 
I'm asking this because I see in my logs a lot of idle connections.

**Comment by apobekiaris** [comment]
it is this module that opens them i have no insights yet as I did not research why this happens

**Comment by CWolffIF** [comment]
I tested it with TCP and no more connections opened.
What is pooling good for?

**Comment by apobekiaris** [comment]
kind of connection manager for connections, XPO guys are taking a look as to why this behaviour, I let u know again

**Comment by apobekiaris** [comment]
The pre-release [4.212.0.4](https://github.com/eXpandFramework/Reactive.XAF/releases/tag/4.212.0.4) in the [Reactive.XAF](https://github.com/eXpandFramework/Reactive.XAF/tree/lab) `lab` branch  includes commits that relate to this task:

* [SequenceGenerator: Connections not closed when DataStorePool #891](https://github.com/eXpandFramework/Reactive.XAF/commit/db928d2ff6aef975ff8a723aa04144e3a97c71a4)

To minimize version conflicts we recommend that you use the [Xpand.XAF.Core.All](https://www.nuget.org/packages/Xpand.XAF.Core.All), [Xpand.XAF.Win.All](https://www.nuget.org/packages/Xpand.XAF.Win.All), [Xpand.XAF.Web.All](https://www.nuget.org/packages/Xpand.XAF.Web.All) packages. Doing so, all packages will be at your disposal and .NET will add a dependecy only to those packages that you actually use and not to all (see the [Modules installation-registrations](https://youtu.be/LvxQ-U_0Sbg) youtube video).
<details>
    <summary>Released packages:</summary>
<a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions'>Xpand.Extensions v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions.Blazor'>Xpand.Extensions.Blazor v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions.Mono.Cecil'>Xpand.Extensions.Mono.Cecil v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions.Office.Cloud'>Xpand.Extensions.Office.Cloud v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions.Office.Cloud.Google.Blazor'>Xpand.Extensions.Office.Cloud.Google.Blazor v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions.Reactive'>Xpand.Extensions.Reactive v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions.XAF'>Xpand.Extensions.XAF v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Extensions/Xpand.Extensions.XAF.Xpo'>Xpand.Extensions.XAF.Xpo v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/tests'>Xpand.TestsLib v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/tests'>Xpand.TestsLib.Blazor v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/tests'>Xpand.TestsLib.Common v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/tests'>Xpand.TestsLib.EasyTest v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/tests'>Xpand.TestsLib.Net461 v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/tools/Xpand.VersionConverter'>Xpand.VersionConverter v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules'>Xpand.XAF.Core.All v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/AutoCommit'>Xpand.XAF.Modules.AutoCommit v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Blazor'>Xpand.XAF.Modules.Blazor v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/CloneMemberValue'>Xpand.XAF.Modules.CloneMemberValue v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/CloneModelView'>Xpand.XAF.Modules.CloneModelView v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Email'>Xpand.XAF.Modules.Email v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/GridListEditor'>Xpand.XAF.Modules.GridListEditor v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/HideToolBar'>Xpand.XAF.Modules.HideToolBar v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/JobScheduler.Hangfire'>Xpand.XAF.Modules.JobScheduler.Hangfire v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/JobScheduler.Notification'>Xpand.XAF.Modules.JobScheduler.Notification v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/JobScheduler.Notification.Email'>Xpand.XAF.Modules.JobScheduler.Notification.Email v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/LookupCascade'>Xpand.XAF.Modules.LookupCascade v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/MasterDetail'>Xpand.XAF.Modules.MasterDetail v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/ModelMapper'>Xpand.XAF.Modules.ModelMapper v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/ModelViewInheritance'>Xpand.XAF.Modules.ModelViewInheritance v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Office.Cloud.Google'>Xpand.XAF.Modules.Office.Cloud.Google v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Office.Cloud.Google.Calendar'>Xpand.XAF.Modules.Office.Cloud.Google.Calendar v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Office.Cloud.Google.Tasks'>Xpand.XAF.Modules.Office.Cloud.Google.Tasks v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Office.Cloud.Microsoft'>Xpand.XAF.Modules.Office.Cloud.Microsoft v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Office.Cloud.Microsoft.Calendar'>Xpand.XAF.Modules.Office.Cloud.Microsoft.Calendar v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Office.Cloud.Microsoft.Todo'>Xpand.XAF.Modules.Office.Cloud.Microsoft.Todo v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Office.DocumentStyleManager'>Xpand.XAF.Modules.Office.DocumentStyleManager v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/OneView'>Xpand.XAF.Modules.OneView v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/PositionInListView'>Xpand.XAF.Modules.PositionInListView v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/ProgressBarViewItem'>Xpand.XAF.Modules.ProgressBarViewItem v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/RazorView'>Xpand.XAF.Modules.RazorView v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Reactive'>Xpand.XAF.Modules.Reactive v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Reactive.Logger'>Xpand.XAF.Modules.Reactive.Logger v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Reactive.Logger.Client.Win'>Xpand.XAF.Modules.Reactive.Logger.Client.Win v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Reactive.Logger.Hub'>Xpand.XAF.Modules.Reactive.Logger.Hub v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Reactive.Rest'>Xpand.XAF.Modules.Reactive.Rest v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/RefreshView'>Xpand.XAF.Modules.RefreshView v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/SequenceGenerator'>Xpand.XAF.Modules.SequenceGenerator v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/SuppressConfirmation'>Xpand.XAF.Modules.SuppressConfirmation v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/ViewEditMode'>Xpand.XAF.Modules.ViewEditMode v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/ViewItemValue'>Xpand.XAF.Modules.ViewItemValue v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/ViewWizard'>Xpand.XAF.Modules.ViewWizard v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules/Windows'>Xpand.XAF.Modules.Windows v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules'>Xpand.XAF.Web.All v.4.212.0.4</a></br> <a href='https://github.com/eXpandFramework/Reactive.XAF/tree/master/src/Modules'>Xpand.XAF.Win.All v.4.212.0.4</a></br>
</details>

Please update the related Nuget packages and test if issues is addressed. These are nightly nuget packages available only from our [NugetServer](https://xpandnugetserver.azurewebsites.net/nuget/).

If you do not use these packages directly but through a module of the main eXpandFramework project, please wait for the bot to notify you again when integration is finished or update the related packages manually.

Thanks a lot for your contribution.

**Comment by apobekiaris** [comment]
Issue is `deprioritized` as `no Assignee found` and scheduled for `auto-close` if no activity in the next `60 days`. Thanks a lot for your contribution.
